### PR TITLE
Enable unit tests for Python3.12

### PR DESCRIPTION
**Pull request by pankajastro**
## Description

This PR enable CI to run unit tests for Python3.12
CI Job: https://github.com/astronomer/astronomer-cosmos/actions/runs/9357004387/job/25755741232?pr=1018

## Related Issue(s)
closes: https://github.com/astronomer/astronomer-cosmos/issues/964

## Checklist

- [ ] I have made corresponding changes to the documentation (if required)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tests/operators/test_base.py [hunk 1]
```diff
@@ -14,6 +15,10 @@
 )
 
 
+@pytest.mark.skipif(
+    (sys.version_info.major, sys.version_info.minor) == (3, 12),
+    reason="The error message for the abstract class instantiation seems to have changed between Python 3.11 and 3.12",
+)
 def test_dbt_base_operator_is_abstract():
```
Comment (pankajastro) [inline tests/operators/test_base.py:22]:
```suggestion
def test_dbt_base_operator_is_abstract_py12():
```

File: tests/operators/test_base.py [hunk 2]
```diff
@@ -23,6 +28,20 @@ def test_dbt_base_operator_is_abstract():
         AbstractDbtBaseOperator()
 
 
+@pytest.mark.skipif(
+    (sys.version_info.major, sys.version_info.minor) != (3, 12),
+    reason="The error message for the abstract class instantiation seems to have changed between Python 3.11 and 3.12",
+)
+def test_dbt_base_operator_is_abstract_py12():
```
Comment (pankajastro) [inline tests/operators/test_base.py:35]:
```suggestion
def test_dbt_base_operator_is_abstract():
```

**Review by tatiana — APPROVED** [review]
This looks great, thanks a lot, @pankajastro !

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -38,11 +38,21 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
         airflow-version: ["2.4", "2.5", "2.6", "2.7", "2.8", "2.9"]
         exclude:
           - python-version: "3.11"
             airflow-version: "2.4"
+          - python-version: "3.12"
```
Comment (tatiana) [inline .github/workflows/test.yml:46]: @pankajastro could you add a comment/link explaining why we're excluding these versions of Airflow?

**Review by pankajkoti — APPROVED** [review]

File: .github/workflows/test.yml (hunk 3 above)
Comment (pankajastro) [reply .github/workflows/test.yml:50]: Added PTAL 29b9f807501724412e9a30984464b4e9b96dbe4f

Comment (tatiana) [reply .github/workflows/test.yml:50]: Thanks, @pankajastro !

File: pyproject.toml [hunk 4]
```diff
@@ -136,7 +136,7 @@ dependencies = [
 pre-install-commands = ["sh scripts/test/pre-install-airflow.sh {matrix:airflow} {matrix:python}"]
 
 [[tool.hatch.envs.tests.matrix]]
-python = ["3.8", "3.9", "3.10", "3.11"]
+python = ["3.8", "3.9", "3.10", "3.11", "3.12"]
 airflow = ["2.4", "2.5", "2.6", "2.7", "2.8", "2.9"]
 
 [tool.hatch.envs.tests.overrides]
```
[the remark this answers is not in the document]
Comment (pankajastro) [reply pyproject.toml:143]: yes, added in a709275cfc1fe54b971161d3f45384b4c8922ad6

**Pull request merged**